sncosmo/salt2utils.pyx

```diff
@@ -215,7 +215,7 @@ cdef class BicubicInterpolator(object):
             y_j = yc[j]
 
             # if y is out of range, we won't be using the value at all
-            if (y_j < self.ymin or y_j > self.ymax):
+            if (not (y_j >= self.ymin and y_j <= self.ymax)):
```

Comment:
Great detective work here! Can you add a comment that we need to take the `yflagvec[j] = -1` branch when `y_j` is NaN? Otherwise someone might unwittingly try to "simplify" this expression back to the original erroneous one.

Reply:
Done!